app/models/manageiq/providers/vmware/infra_manager/vm/reconfigure.rb

```diff
@@ -52,6 +52,118 @@ def build_config_spec(task_options)
       end
       set_spec_option(vmcs, :memoryMB, task_options[:vm_memory],      :to_i)
       set_spec_option(vmcs, :numCPUs,  task_options[:number_of_cpus], :to_i)
+
+      if task_options[:disk_remove] || task_options[:disk_add]
+        with_provider_object do |vim_obj|
+          task_options[:disk_remove].each { |d| remove_disk_config_spec(vim_obj, vmcs, d) } if task_options[:disk_remove]
+          add_disks(vim_obj, vmcs, task_options[:disk_add]) if task_options[:disk_add]
+        end
+      end
+    end
+  end
+
+  def add_disks(vim_obj, vmcs, disks)
+    ck, un = vim_obj.send(:getScsiCandU)
+
+    # if there is no scsi controller
+    if ck.nil?
+      ck, un = [-99, 0]
+      add_scsi_controller(vmcs, 0, ck)
+    end
+
+    disks.each do |d|
+      d[:controller_key] = ck
+      d[:unit_number]    = un
+      add_disk_config_spec(vmcs, d)
+      un += 1
+    end
+  end
+
+  def add_scsi_controller(vmcs, bus_number, dev_key)
+    device_type = 'VirtualLsiLogicController'
+    add_device_config_spec(vmcs, VirtualDeviceConfigSpecOperation::Add) do |vdcs|
+      vdcs.device = VimHash.new(device_type) do |vDev|
+        vDev.sharedBus = VimString.new('noSharing', 'VirtualSCSISharing')
+        vDev.busNumber = bus_number
+        vDev.key       = dev_key
+      end
+    end
+  end
+
+  def backing_filename(diskname, unit_number)
+    filename = (diskname.blank? ? "disk_#{Time.zone.now.iso8601.gsub(/[-:TZ]/, '')}" : diskname) + "_#{unit_number}.vmdk"
+
+    folder, datastore = File.dirname(location).split.reverse
```

Comment:
@lfu Is this really doing what you want?  `location` does not contain the datastore name and splitting based on a space in the name is not correct either has the paths can contain spaces.

It might be better to look at the disk objects attached through the `hardware` relationship to find the target datastore and fall back to the datastore of the configuration file `storage.name` if we do not find a suitable disk.